### PR TITLE
Add VSCode Recommendations and Basic Settings

### PR DESCRIPTION
**Pull request by panicstevenson**
Initial VSCode extension recommendations (basically just ESLint and @MaikoTan's highlighting extension from https://github.com/quisquous/cactbot/issues/1846), as well as some simple settings.

File associations won't be upset if the cactbot-highlight extension isn't installed. I don't know if there's any reason to have the `.git` folder appear in VSCode (potentially maybe node_modules can be here too?). Likewise, I don't think there's much point in search results showing dist or node_modules.

Perhaps these can be extended to include third-party files, such as lodash.min.js, etc. but I figured it might be best to ground something small first.

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  // See https://go.microsoft.com/fwlink/?LinkId=827846
```
Comment (quisquous) [inline .vscode/extensions.json:2]: github syntax highlighting doesn't like these comments, but I think they are also valid here.  (Trailing commas also appear to be valid?? :eyes:)

Do you think it's worth adding this line to the `.gitattributes` file?
```
.vscode/*.json jsonc
```

Comment (panicstevenson) [reply .vscode/extensions.json:2]: We can certainly try and find out what happens!
I just copied/pasted it from other people; it seems like most everyone has these comments in (despite them being so ugly in GitHub)

Comment (panicstevenson) [reply .vscode/extensions.json:2]: I think GitHub still hates it :(

Comment (panicstevenson) [reply .vscode/extensions.json:2]: GitHub likes this now :)

**Pull request merged**